src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,129 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegisterException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void successfullyUserRegister_Ok() {
+        User user = new User();
+        user.setLogin("test login");
+        user.setPassword("password");
+        user.setAge(20);
+
+        User register = registrationService.register(user);
+        assertEquals(user, register);
+        assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void userAlreadyExists_NotOk() {
+        User user = new User();
+        user.setLogin("existing login");
+        user.setAge(20);
+        user.setPassword("password");
+        Storage.people.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("existing login");
+        newUser.setPassword("new user password");
+        newUser.setAge(19);
+
+        assertThrows(RegisterException.class,
+                () -> registrationService.register(newUser),
+                "Login already exists! Enter another value");
+    }
+
+    @Test
+    void userIsNull_NotOk() {
+        assertThrows(RegisterException.class,
```

Comment:
The expected message in the `assertThrows` method for the null user test should match the message thrown in the `RegistrationServiceImpl`. The message should be "User can not be null" instead of "User can not be null!" to match the actual exception message.

Reply:
fixed